cmd/manager/main.go

```diff
@@ -674,6 +683,58 @@ func chooseAndValidateIPFamily(ipFamilyStr string, ipFamilyDefault corev1.IPFami
 	}
 }
 
+// determineSetDefaultSecurityContext determines what settings we need to use for security context by using the following rules:
+// 1. If the setDefaultSecurityContext is explicitly set to either true, or false, use this value.
+// 2. use openshift detection pulled from kubevirt
+//    https://github.com/kubevirt/kubevirt/blob/f71e9c9615a6c36178169d66814586a93ba515b5/pkg/util/cluster/cluster.go#L21
+//    to determine whether or not we are running within an openshift cluster.  If we determine we are on an openshift cluster,
+//    then default to false, otherwise, return true.
+func determineSetDefaultSecurityContext(setDefaultSecurityContext string, clientset kubernetes.Interface) (bool, error) {
+	isSet, err := strconv.ParseBool(setDefaultSecurityContext)
+	if err == nil {
+		return isSet, nil
+	}
+	_, apiResources, err := clientset.Discovery().ServerGroupsAndResources()
+	if err != nil && !discovery.IsGroupDiscoveryFailedError(err) {
+		return false, err
+	}
+
+	openshiftSecurityGroupName := "security.openshift.io"
+	openshiftSecurityGroupVersion := schema.GroupVersion{Group: openshiftSecurityGroupName, Version: "v1"}
+
+	// In case of an error, check if security.openshift.io is the reason (unlikely).
+	// If it is, we are obviously on an openshift cluster.
+	if discovery.IsGroupDiscoveryFailedError(err) {
+		var e *discovery.ErrGroupDiscoveryFailed
+		if ok := errors.As(err, &e); ok {
+			if _, exists := e.Groups[openshiftSecurityGroupVersion]; exists {
+				// since we have determined we are absolutely running within an openshift cluster,
+				// behave as if "setDefaultSecurityContext" was set to false.
+				return false, nil
+			}
+		}
+	}
+
+	// search for "securitycontextconstraints" within the cluster's api resources,
+	// since this is an openshift specific api resource that does not exist outside of openshift.
+	for _, apiResource := range apiResources {
+		if apiResource.GroupVersion == openshiftSecurityGroupVersion.String() {
+			for _, resource := range apiResource.APIResources {
+				if resource.Name == "securitycontextconstraints" {
+					// since we have determined we are absolutely running within an openshift cluster,
+					// behave as if "setDefaultSecurityContext" was set to false.
+					return false, nil
+				}
+			}
+		}
+	}
```

Comment:
Slightly shorter / optimized version:

```suggestion
	openshiftSecurityGroupVersion := schema.GroupVersion{Group: "security.openshift.io", Version: "v1"}
	apiResourceList, err := clientset.Discovery().ServerResourcesForGroupVersion(openshiftSecurityGroupVersion.String())
	if err != nil {
		// In case of an error, check if security.openshift.io is the reason (unlikely).
		var e *discovery.ErrGroupDiscoveryFailed
		if ok := errors.As(err, &e); ok {
			if _, exists := e.Groups[openshiftSecurityGroupVersion]; exists {
				// since we have determined we are absolutely running within an openshift cluster,
				// behave as if "setDefaultSecurityContext" was set to false.
				return false, nil
			}
		}
		return false, err
	}

	// search for "securitycontextconstraints" within the cluster's api resources,
	// since this is an openshift specific api resource that does not exist outside of openshift.
	for _, apiResource := range apiResourceList.APIResources {
		if apiResource.Name == "securitycontextconstraints" {
			// since we have determined we are absolutely running within an openshift cluster,
			// behave as if "setDefaultSecurityContext" was set to false.
			return false, nil
		}
	}
```